shakenfist/images.py

```diff
@@ -30,6 +31,32 @@
     'ubuntu': image_resolver_ubuntu
 }
 
+IMAGE_FETCH_LOCK_TIMEOUT = 600   # TODO(andy):Should be linked to HTTP timeout?
+
+
+def get_image(url, locks, op_label, timeout=IMAGE_FETCH_LOCK_TIMEOUT):
+    """Fetch image if not downloaded and return image path"""
+    hashed = hash_image_url(url)
+    image_lock_name = 'sf/images/%s/%s' % (
```

Comment:
I worry that we're quite exposed to a lock name typo as well. Maybe we need a path generator helper like we have for data in etcd?

Reply:
yes I agree. Probably a separate PR due to the many changes required.
I do have further plans for this lock. My next PR will be to make this lock per node. (Next time I'm actually working on SF.)